runtime/src/main/scala/akka/grpc/javadsl/Metadata.scala

```diff
@@ -50,13 +50,17 @@ trait Metadata {
 }
 
 /**
+ * Provides access to details to more rich error details using the logical gRPC com.google.rpc.Status message, see
+ * [API Design Guide](https://cloud.google.com/apis/design/errors) for more details.
+ *
  * Not for user extension
  */
 @ApiMayChange
 @DoNotInherit
-trait RichMetadata extends Metadata {
+trait MetadataStatus extends Metadata {
+  def getStatus(): com.google.rpc.Status
   def getCode(): Int
   def getMessage(): String
   def getDetails(): List[com.google.protobuf.any.Any]
-  def getParsedDetails[K <: scalapb.GeneratedMessage](index: Int, companion: scalapb.GeneratedMessageCompanion[K]): K
+  def getParsedDetails[K <: scalapb.GeneratedMessage](companion: scalapb.GeneratedMessageCompanion[K]): List[K]
```

Comment:
Likely contain only one of each type, but it's also possible with many, LocalizedMessage would be one per language for example.